api/v1alpha1/etcdcluster_types.go

```diff
@@ -17,23 +17,18 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"k8s.io/apimachinery/pkg/api/resource"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type Storage struct {
-	StorageClass string            `json:"storageClass"`
-	Size         resource.Quantity `json:"size"`
-}
-
 // EtcdClusterSpec defines the desired state of EtcdCluster
 type EtcdClusterSpec struct {
 	// Replicas is the count of etcd instances in cluster.
 	// +optional
 	// +kubebuilder:default:=3
 	// +kubebuilder:validation:Minimum:=0
-	Replicas *int32  `json:"replicas,omitempty"`
-	Storage  Storage `json:"storage,omitempty"`
+	Replicas *int32      `json:"replicas,omitempty"`
+	Storage  StorageSpec `json:"storage"`
```

Comment:
The new spec looks good good. The onlyquestion if we do this:

```
storage:
   volumeClaimTemplate:
     metadata:
       labels:
         env:prod
       annotations:
         example.com/annotation: "true"
     spec: # core.v1.PersistentVolumeClaimSpec Ready k8s type
       storageClassName: gp3
       accessModes: ["ReadWriteOnce"]
       resources:
         requests:
           storage: 10Gi
   emptyDir: {} # core.v1.EmptyDirVolumeSource Ready k8s type
```

then where will we add the possible options in the future:
1) auto-deletion of PVC following the EtcdCluster object
2) options for resizing PVC (resizeInUseVolumes as CNPG and mariadb-operator do
3) option to manage storage migration
4) any other options for storage in the future

Reply:
Okay, I'll move this into a new issue, LGTM for now